sources/packages/backend/apps/api/src/services/student-scholastic-standings/student-scholastic-standings.service.ts

```diff
@@ -283,81 +300,140 @@ export class StudentScholasticStandingsService extends RecordDataModelService<St
       scholasticStandingData.scholasticStandingChangeType ===
       StudentScholasticStandingChangeType.StudentDidNotCompleteProgram
     ) {
+      // Only check for SSR if SSR or SSRN are not present.
       if (!scholasticStandingData.numberOfUnsuccessfulWeeks) {
         throw new Error("Number of unsuccessful weeks is empty.");
       }
       const totalExistingUnsuccessfulWeeks =
         await this.getTotalFullTimeUnsuccessfulWeeks(studentId);
-
       // When total number of unsuccessful weeks hits minimum 68, add SSR restriction.
       if (
         totalExistingUnsuccessfulWeeks +
           scholasticStandingData.numberOfUnsuccessfulWeeks >=
         SCHOLASTIC_STANDING_MINIMUM_UNSUCCESSFUL_WEEKS
       ) {
-        return this.studentRestrictionSharedService.createRestrictionToSave(
-          studentId,
-          RestrictionCode.SSR,
-          auditUserId,
-          applicationId,
-        );
+        const ssrRestriction =
+          await this.studentRestrictionSharedService.createRestrictionToSave(
+            studentId,
+            RestrictionCode.SSR,
+            auditUserId,
+            applicationId,
+          );
+        restrictions.push(ssrRestriction);
+        return restrictions;
       }
     }
     if (
       scholasticStandingData.scholasticStandingChangeType ===
       StudentScholasticStandingChangeType.StudentWithdrewFromProgram
     ) {
+      // Check for "WTHD" restriction since it should always be created if one is not present and active.
+      const hasActiveWTHD = existingRestrictions.some(
+        (studentRestriction) =>
+          studentRestriction.restriction.restrictionCode ===
+            RestrictionCode.WTHD && studentRestriction.isActive,
+      );
       // Check if "WTHD" restriction is already present for the student,
-      // if not add "WTHD" restriction else add "SSR" restriction.
-      const isWTHDAlreadyExists =
-        await this.studentRestrictionService.hasAnyActiveRestriction(
-          studentId,
-          [RestrictionCode.WTHD],
-        );
-
-      const restrictionCode = isWTHDAlreadyExists
-        ? RestrictionCode.SSR
-        : RestrictionCode.WTHD;
-
-      return this.studentRestrictionSharedService.createRestrictionToSave(
-        studentId,
-        restrictionCode,
-        auditUserId,
-        applicationId,
+      // if not, then create a new one.
+      if (!hasActiveWTHD) {
+        const wthdRestriction =
+          await this.studentRestrictionSharedService.createRestrictionToSave(
+            studentId,
+            RestrictionCode.WTHD,
+            auditUserId,
+            applicationId,
+          );
+        restrictions.push(wthdRestriction);
+      }
+      // Check if an SSR or SSRN was created for the student.
+      const createdSSROrSSRN = restrictions.some((studentRestriction) =>
+        [RestrictionCode.SSR, RestrictionCode.SSRN].includes(
+          studentRestriction.restriction.restrictionCode as RestrictionCode,
+        ),
```

Comment:
the variable `hasEscalationRestrictions` can be re-used here?

![image](https://github.com/user-attachments/assets/84d5c75f-4957-42bd-8a89-ae5cbc9def15)

Reply:
The check is performed in different arrays.